src/main/java/org/usfirst/frc/team4099/robot/subsystems/Drive.kt

```diff
@@ -30,16 +49,125 @@ class Drive private constructor() : Subsystem {
         leftSlave1SRX.set(ControlMode.Follower, Constants.Drive.LEFT_MASTER_ID.toDouble())
         leftSlave2SRX.set(ControlMode.Follower, Constants.Drive.LEFT_MASTER_ID.toDouble())
         leftMasterSRX.set(ControlMode.PercentOutput, 0.0)
+        leftMasterSRX.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder, 0, 10)
+        leftMasterSRX.setSensorPhase(false)
+        leftMasterSRX.setStatusFramePeriod(StatusFrameEnhanced.Status_2_Feedback0, 5, 10)
+        leftMasterSRX.config_kP(0, Constants.Gains.LEFT_LOW_KP, 10)
+        leftMasterSRX.config_kI(0, Constants.Gains.LEFT_LOW_KI, 10)
+        leftMasterSRX.config_kD(0, Constants.Gains.LEFT_LOW_KD, 10)
+        leftMasterSRX.config_kF(0, Constants.Gains.LEFT_LOW_KF, 10)
+
+        leftMasterSRX.config_kP(1, Constants.Gains.LEFT_HIGH_KP, 10)
+        leftMasterSRX.config_kI(1, Constants.Gains.LEFT_HIGH_KI, 10)
+        leftMasterSRX.config_kD(1, Constants.Gains.LEFT_HIGH_KD, 10)
+        leftMasterSRX.config_kF(1, Constants.Gains.LEFT_HIGH_KF, 10)
+
+
 
         rightSlave1SRX.set(ControlMode.Follower, Constants.Drive.RIGHT_MASTER_ID.toDouble())
         rightSlave2SRX.set(ControlMode.Follower, Constants.Drive.RIGHT_MASTER_ID.toDouble())
         rightMasterSRX.set(ControlMode.PercentOutput, 0.0)
+        rightMasterSRX.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder, 0, 10)
+        rightMasterSRX.setSensorPhase(false)
+        rightMasterSRX.setStatusFramePeriod(StatusFrameEnhanced.Status_2_Feedback0, 5, 10)
+        rightMasterSRX.config_kP(0, Constants.Gains.RIGHT_LOW_KP, 10)
+        rightMasterSRX.config_kI(0, Constants.Gains.RIGHT_LOW_KI, 10)
+        rightMasterSRX.config_kD(0, Constants.Gains.RIGHT_LOW_KD, 10)
+        rightMasterSRX.config_kF(0, Constants.Gains.RIGHT_LOW_KF, 10)
+
+        rightMasterSRX.config_kP(1, Constants.Gains.RIGHT_HIGH_KP, 10)
+        rightMasterSRX.config_kI(1, Constants.Gains.RIGHT_HIGH_KI, 10)
+        rightMasterSRX.config_kD(1, Constants.Gains.RIGHT_HIGH_KD, 10)
+        rightMasterSRX.config_kF(1, Constants.Gains.RIGHT_HIGH_KF, 10)
+
+        leftMasterSRX.configVelocityMeasurementPeriod(VelocityMeasPeriod.Period_10Ms,10)
+        leftMasterSRX.configVelocityMeasurementWindow(32, 10)
+        rightMasterSRX.configVelocityMeasurementPeriod(VelocityMeasPeriod.Period_10Ms,10)
+        rightMasterSRX.configVelocityMeasurementWindow(32,10)
+
+        setOpenLoop(DriveSignal.NEUTRAL)
+
+
 
         ahrs = AHRS(SPI.Port.kMXP)
 
         this.zeroSensors()
     }
 
+
+    @Synchronized
+    fun setOpenLoop(signal: DriveSignal) {
+        if (currentState !== DriveControlState.OPEN_LOOP) {
+            leftMasterSRX.set(ControlMode.PercentOutput, 0.0)
+            rightMasterSRX.set(ControlMode.PercentOutput, 0.0)
+            leftMasterSRX.configNominalOutputForward(0.0, 10)
```

Comment:
I really don't like the timeout thing -- apparently, it's blocking and will hold the thread until it gets a response. It may be better to just give it 0 timeout and hope it went through. Is there ever a time where we change it from 0.0 to anything? If not, just give it zero timeout.

Reply:
So should I set all timeouts to 0?